pkg/metrics/server.go

```diff
@@ -4,28 +4,57 @@ import (
 	"bytes"
 	"context"
 	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
 	"errors"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"net/http"
 	"os"
 	"time"
 
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/cluster-node-tuning-operator/pkg/util"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"gopkg.in/fsnotify.v1"
-
-	"k8s.io/klog/v2"
 )
 
 const (
 	tlsSecretDir = "/etc/secrets"
 	tlsCert      = tlsSecretDir + "/tls.crt"
 	tlsKey       = tlsSecretDir + "/tls.key"
+
+	authCADir                = "/tmp/metrics-client-ca"
+	authCAFile               = authCADir + "/ca.crt"
+	AuthConfigMapNamespace   = "kube-system"
+	AuthConfigMapName        = "extension-apiserver-authentication"
+	AuthConfigMapClientCAKey = "client-ca-file"
 )
 
 type Server struct {
 }
 
+// DumpCA writes the root certificate bundle which is used to verify client certificates
+// on incoming requests to 'authCAFile' file.
+func DumpCA(ca string) error {
+	if err := util.Mkdir(authCADir); err != nil {
+		return fmt.Errorf("failed to create directory %q: %v", authCADir, err)
+	}
+
+	f, err := os.Create(authCAFile)
+	if err != nil {
+		return fmt.Errorf("failed to create file %q: %v", authCAFile, err)
+	}
+	defer f.Close()
+	if _, err = f.WriteString(ca); err != nil {
```

Comment:
Creating and writing to the file might generate several fsnotify events which could trigger several recreations of the server?

Reply:
I believe this is very unlikely.  I was able to write a 1M test file with a single WRITE event.  Also, inotify does only fire up after a write is completed.  What will happen though is two WRITE events when the `authCAFile` already exists.  The first one will create an empty file via `os.Create()` and the second WRITE will write the actual data.  However, we account for empty files in the code so no extra recreations of the serve will happen.